src/test/java/teammates/test/AbstractBackDoor.java

```diff
@@ -260,11 +261,14 @@ public void removeDataBundle(DataBundle dataBundle) {
     /**
      * Puts searchable documents in data bundle into the datastore.
      */
-    public String putDocuments(DataBundle dataBundle) {
+    public String putDocuments(DataBundle dataBundle) throws HttpRequestFailedException {
```

Comment:
Why not return a boolean here instead of throwing an exception?

Reply:
To make it the same as `removeAndRestoreDataBundle`. The latter needs to be `String`-returning as there is a usage in L&P test case that requires it to be. And I do think there might be future usage of `putDocuments` that requires the output result, but I could be wrong.

Reply:
Well fair enough, if it makes the codebase more consistent we can retain this approach. I believe we should however, make the happy path prominent:

```
        if (putRequestOutput.responseCode != HttpStatus.SC_OK) {
            throw new HttpRequestFailedException("Request failed: [" + putRequestOutput.responseCode + "] "
                + putRequestOutput.responseBody);
        }
        
        return putRequestOutput.responseBody;
```


It helps to make the code more readable